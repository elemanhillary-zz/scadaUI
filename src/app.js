import React, { Component } from 'react';
import { connect } from 'react-redux';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import {
  LoginPage, SignupPage,
  MyDropzone, PrivateRoute,
} from './components';
import { alertActions } from './redux/actions';

class App extends Component {
  constructor(props) {
    super(props);

    const { dispatch } = this.props;
    dispatch(alertActions.clear());
  }

  render() {
    const { alert } = this.props;
    return (
      <div className="jumbotron">
        <div className="container">
          <div className="col-sm-8 col-sm-offset-2">
            {alert.message
              && <div className={`alert ${alert.type}`}>{alert.message}</div>
            }
            <Router>
              <div>
                <PrivateRoute exact path="/" component={MyDropzone} />
                <Route path="/signin" component={LoginPage} />
                <Route path="/signup" component={SignupPage} />
              </div>
            </Router>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { alert } = state;
  return {
    alert,
  };
}

const connectedApp = connect(mapStateToProps)(App);
export { connectedApp as App };

export default App;
