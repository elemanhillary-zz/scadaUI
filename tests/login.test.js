/* eslint-disable no-undef */
import React from 'react';
import { shallow } from './helpers/config';
import { LoginPage, SignupPage } from '../src/components';

describe('<Login />', () => {
  let wrapper;
  const props = {
    login: jest.fn(),
    username: '',
    password: '',
    errors: {},
    loggingIn: false,
  };
  beforeEach(() => {
    wrapper = shallow(<LoginPage {...props} />);
  });
  it('should render Login component', () => {
    wrapper = shallow(<LoginPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<Signup />', () => {
  let wrapper;
  const props = {
    signin: jest.fn(),
    username: '',
    password: '',
    errors: {},
    registering: false,
  };
  beforeEach(() => {
    wrapper = shallow(<SignupPage {...props} />);
  });
  it('should render SignupPage component', () => {
    wrapper = shallow(<SignupPage {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
