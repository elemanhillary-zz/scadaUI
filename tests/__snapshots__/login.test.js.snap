// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render Login component 1`] = `ShallowWrapper {}`;

exports[`<Signup /> should render SignupPage component 1`] = `ShallowWrapper {}`;
